### ISSUE TITLE
Single process specific bug

### ISSUE THREAD

**Issue by Feodor2** [Feodor2/Mypal68#466]
Так вот я докрутил али тот до конца, да с 900мб начло тормозить, всего дошло до 1100 примерно, закрываю освободило до 900, конечно это не дело, делаю measure в about:memory, видно что али сидит себе как-будто его и не закрывали, то есть что то мешает ему закрыться и освободить память.

Для начала нужно выяснить с какой версии появилась эта проблема.

Кстати в мультипроцессе такой проблемы нет - закрыл , закрылся процесс и память освободилась.

**Comment by NS-Clone** [comment]
> видно что али сидит себе как-будто его и не закрывали, то есть что то мешает ему закрыться и освободить память

у майпалов скорей всего с самых первых

**Comment by bluedragon-cairo** [comment]
This happens not just in Mypal68 but also in New Moon by roytam1 and your old PM based Mypal too. It is annoying and requires me to restart the browser when it hits 1GB. The browser starts to occupy all the CPU

**Comment by Ariel-Schnee** [comment]
> This happens not just in Mypal68 but also in New Moon by roytam1 and your old PM based Mypal too. It is annoying and requires me to restart the browser when it hits 1GB. The browser starts to occupy all the CPU when high RAM

This has been happening on Firefox as far back as Firefox official XP builds. It's something Mozilla's been told about repeatedly, but have never fixed. And they're not likely to anytime either.

**Comment by NS-Clone** [comment]
> It is annoying and requires me to restart the browser when it hits 1GB.

that's why i'm using multiprocess
it eat MOAR ram
but it works when browser eat 3GB

**Comment by bluedragon-cairo** [comment]
> that's why i'm using multiprocess

I use New Moon more these days because XUL is cool & WebExtensions is a garbage - although Mypal68 is normally faster than New Moon (there are some web pages that New moon is faster than Mypal68 tho)

**Comment by Feodor2** [comment]
Yes old firefox based have this problem, do not release much memory.
But not with all sites, let we find what exact site things cause this.

**Comment by NS-Clone** [comment]
> New Moon

i think NM is slightly "outdated"
serpent52 runs MOAR firefox extensions

**Comment by NS-Clone** [comment]
> are you using XP 64bit?

nope
i'm using 4GB pagefile on SSD

**Comment by bluedragon-cairo** [comment]
> hmmm does old x64 firefox works well on x64 XP?

not sure as I never used XP x64 on real hardware

> but browser works well when it eat 2GB

I couldn't make the browser eat 2gb because even 1.2GB is the browser almost impossible to use & couldn't do anything. just a note-My 2 XP PCs has 8GB and 12GB physical RAM each with PAE enable patched kernel.

> i think NM is slightly "outdated" / serpent52 runs MOAR useful firefox addons

I just liked the full customizability with early FF UI. Classic theme restorer(for serpent/centaury) is slightly limited

**Comment by NS-Clone** [comment]
> I couldn't make the browser eat 2gb because even 1.2GB is the browser almost impossible to use & couldn't do anything

it's about single process
2GB process memory gets fragmented and...

in multiprocess it's works like firefox with used only 600MB :)
![2024-06-15_063158](https://github.com/Feodor2/Mypal68/assets/139008911/684689ed-0347-4b30-9b84-61206d540301)

**Comment by Feodor2** [comment]
After the observation, I found 4.4 version does not free memory on closed site too also crashes of out memory at 1.5gb even if is free memory available.

**Comment by bluedragon-cairo** [comment]
(untested) Try getting older Firefox or Pale Moon and install an extension called "Memory Fox", and in the toolbar/status bar menu select "Activate Memory Fox - All Processes". This extension will control memory leak from all processes (outside the browser too) so I think Mypal68 will be effected too. (Since XUL based extensions cannot be installed directly to Mypal68)

**Comment by Feodor2** [comment]
Check the new 14.5 version
